src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -138,36 +150,89 @@ private boolean compareResource(Resource requestResource, String policyResource)
         return ANY_REGEX.equals(policyResource);
     }
 
-    private Operation parseOperation(String operationStr) {
+    private Operation parseOperation(String operationStr) throws PolicyException {
         if (Utils.isEmpty(operationStr)) {
-            throw new IllegalArgumentException("Operation can't be empty");
+            throw new PolicyException("Operation can't be empty");
+        }
+
+        int split = operationStr.indexOf(':');
+        if (split == -1) {
+            throw new PolicyException(EXCEPTION_MALFORMED_OPERATION);
+        }
+
+        String service = operationStr.substring(0, split);
+        if (service.isEmpty() || !StringUtils.isAlpha(service)) {
+            throw new PolicyException(EXCEPTION_MALFORMED_OPERATION);
         }
 
-        Matcher matcher = SERVICE_OPERATION_PATTERN.matcher(operationStr);
-        if (matcher.matches()) {
-            return Operation.builder().operationStr(operationStr).service(matcher.group(1)).action(matcher.group(2))
-                    .build();
+        String action = operationStr.substring(split + 1);
```

Comment:
this can throw if split is at the end of the string

Reply:
😓

Reply:
this can still throw, I don't think you're doing any validation